src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -180,6 +187,11 @@ class PythonSetup(Subsystem):
             """
         ),
     )
+    pip_version = EnumOption(
+        default=PipVersion.V22_3,
```

Comment:
This is potentially controversial, as we're also upgrading everyone to newer pip, rather than making it opt-in.

I think that's fine, since pip is very heavily tested, and ubiquitous, and we want to bias towards doing the right thing for new users who will not be expecting to be going back to some ancient pip. 

We have an escape hatch for those who encounter an unexpected and unlikely problem in an upgrade.